corehq/apps/users/middleware.py

```diff
@@ -43,16 +43,26 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if 'org' in view_kwargs:
             request.org = view_kwargs['org']
         if request.user and request.user.is_authenticated:
-            user_id = username_to_user_id(request.user.username)
-            request.couch_user = CouchUser.get_by_user_id(user_id)
+            couch_user = _get_couch_user(request.user.username)
+            if not couch_user:
+                couch_user = InvalidUser()
+            request.couch_user = couch_user
             if not request.couch_user.analytics_enabled:
                 request.analytics_enabled = False
             if 'domain' in view_kwargs:
                 domain = request.domain
-                if not request.couch_user:
-                    request.couch_user = InvalidUser()
-                if request.couch_user:
-                    request.couch_user.current_domain = domain
+                request.couch_user.current_domain = domain
         elif is_public_reports(view_kwargs, request):
             request.couch_user = AnonymousCouchUser()
         return None
+
+
+def _get_couch_user(username):
```

Comment:
Since there's a `CouchUser.get_by_user_id`, maybe this should also be moved there and renamed so that it's `CouchUser.get_by_username`? Could be useful in other places

Reply:
I re-thought this after getting some more info about one of the places it was happening.

Reply:
Oh I see, let me know if it's ready for review again